test/e2eTests/tests/patientSearch_test.js

```diff
@@ -0,0 +1,78 @@
+// Test for:
+// https://docs.google.com/document/d/1hihKCjCQVprju_SCi1nqXCYchY-ThX4iqHPkv4asnt0/edit#heading=h.gpe6dgjlwx02
+
+Feature('Patient Search');
+
+Before(async (I, Apis, Data) => {
+	// Create the patients
+	I.say("Creating patient one")
+	await Apis.patient.create(Data.users.patient1)
+	
+	I.say("Creating patient two")
+	await Apis.patient.create(Data.users.patient2)
+
+	// Login
+  I.login()
+})
+
+Scenario('Validate searching works on the clinic page', (I, Apis, Data, DashboardPage) => {
+	const clinicPage = DashboardPage.navigateToClinicPage()
+	validateSearch(I, Data, clinicPage)
+})
+
+Scenario('Validate searching works on the lab page', (I, Apis, Data, DashboardPage) => {
+	const labPage = DashboardPage.navigateToLabPage()
+	validateSearch(I, Data, labPage)
+})
+
+Scenario('Validate searching works on the pharmacy page', (I, Apis, Data, DashboardPage) => {
+	const pharmacyPage = DashboardPage.navigateToPharmacyPage()
+	validateSearch(I, Data, pharmacyPage)
+})
+
+Scenario('Validate searching works on the registration page', (I, Apis, Data, DashboardPage) => {
+	const registrationPage = DashboardPage.navigateToRegistrationPage()
+	validateSearch(I, Data, registrationPage)
+})
+
+Scenario('Validate searching works on the social page', (I, Apis, Data, DashboardPage) => {
+	const socialPage = DashboardPage.navigateToSocialPage()
+	validateSearch(I, Data, socialPage)
+})
+
+Scenario('Validate searching works on the vitals page', (I, Apis, Data, DashboardPage) => {
+	const vitalsPage = DashboardPage.navigateToVitalsPage()
+	validateSearch(I, Data, vitalsPage)
+})
+
+// Runs the core search scenario on the given page
+const validateSearch = (I, Data, pageWithSearch) => {
```

Comment:
This is what it's all been about: a single function that runs the same set of test logic on different pages. Callers navigate to the page they are testing then call this function to run the checks, so we can reuse this function for each test. ```validateSearch``` just expects the passed in ```Page``` to have search related functions, like ```seach()``` and ```seePatientRecord()```, but it doesn't know or care that they were actually copied from the ```Page```'s ```Component```